app/passwordValidator.js

```diff
@@ -0,0 +1,67 @@
+function validate(password, ruleList) {
+  let errors = [];
+  ruleList.forEach(function(ruleName) {
+    let validationRule = validationRules[ruleName];
+    if (validationRule != null) {
+      let error = validationRule(password);
+      if (error != null) {
+        errors.push(error);
+      }
+    }
+  });
+
+  return errors;
+}
+
+class ValidationError {
+  constructor(translationKey, args) {
+    this.translationKey = translationKey;
+    this.args = args;
+  }
+}
+
+function maxLengthValidate(length) {
+  return function(password) {
+    if (password.length >= length) {
+      return new ValidationError('maxPasswordLength', {
+        length
+      });
+    }
+    return null;
+  };
+}
+
+function minLengthValidate(length) {
+  return function(password) {
+    if (password.length < length) {
+      return new ValidationError('minPasswordLength', {
+        length
+      });
+    }
+    return null;
+  };
+}
+
+function containsCharacterClass(regex, count, classDescription) {
+  return function(password) {
+    let matches = password.match(regex);
+    if (matches == null || matches.length < count) {
+      return new ValidationError('characterClass', {
+        classDescription,
+        count
+      });
+    }
+    return null;
+  };
+}
+
+let validationRules = {
+  length_32: maxLengthValidate(32),
```

Comment:
Works for our use-case, but if this is going upstream shouldn't configuration support arbitrary numbers for each rule? I think node-convict could support this with a custom or object format.

Reply:
Hm, @gusk also suggested something like this one: https://validatejs.org/ .  Looking at that it seems kind of complicated; I'd like it to still be configurable with a simple environment variable.  I think you're right that the current implementation is too rigid.

Reply:
What if we split each rule string on `_` and `parseInt` the second element?

Reply:
I added a nested object to convict. I couldn't figure out how to get convict to parse an object from an environment variable, so I added a config.json file to put in this and any other config.  Please have another look @cjford :)